code/_onclick/ai.dm

```diff
@@ -190,24 +190,37 @@
 	else if(down)
 		TD.move_camera_by_click()
 
-/turf/AIShiftClick(mob/living/silicon/ai/user)
+
+///AI Aiming. This is sending info to a linked mortar/howi
+/atom/proc/ai_artillery_aim(mob/living/silicon/ai/user, target)
 	if(!user.linked_artillery)
 		to_chat(user, span_notice("No linked mortar found."))
 		return
 	
-	var/area/A = get_area(src)
+	var/area/A = get_area(target)
 	if(istype(A) && A.ceiling >= CEILING_UNDERGROUND)
 		to_chat(user, span_warning("You cannot hit the target. It is probably underground."))
 		return
 	to_chat(user, span_notice("Sending targeting information to [user.linked_artillery]. COORDINATES: X:[x] Y:[y]"))
-	user.linked_artillery.recieve_target(src,user)
+	user.linked_artillery.recieve_target(target, user)
+
+/turf/AIShiftClick(mob/living/silicon/ai/user)
+	ai_artillery_aim(user, src)
```

Comment:
Just make a signal on the AI for this instead of puttin gai altillery aim on like everythin in the game

Reply:
I doubt I will merge it in this state then but I might talk about it with brave

Reply:
I already said no to small about that on discord.